Document Materialize MCP Server

sjwiesman opened the pull request:
<!--
Describe the contents of the PR briefly but completely.

If you write detailed commit messages, it is acceptable to copy/paste them
here, or write "see commit messages for details." If there is only one commit
in the PR, GitHub will have already added its commit message above.
-->

### Motivation

Preview: https://preview.materialize.com/materialize/32426/integrations/llm/

<!--
Which of the following best describes the motivation behind this PR?

  * This PR fixes a recognized bug.

    [Ensure issue is linked somewhere.]

  * This PR adds a known-desirable feature.

    [Ensure issue is linked somewhere.]

  * This PR fixes a previously unreported bug.

    [Describe the bug in detail, as if you were filing a bug report.]

  * This PR adds a feature that has not yet been specified.

    [Write a brief specification for the feature, including justification
     for its inclusion in Materialize, as if you were writing the original
     feature specification.]

   * This PR refactors existing code.

    [Describe what was wrong with the existing code, if it is not obvious.]
-->

### Tips for reviewer

<!--
Leave some tips for your reviewer, like:

    * The diff is much smaller if viewed with whitespace hidden.
    * [Some function/module/file] deserves extra attention.
    * [Some function/module/file] is pure code movement and only needs a skim.

Delete this section if no tips.
-->

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  <!-- Reference the design in the description. -->
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  <!-- Ask in #team-cloud on Slack if you need help preparing the cloud PR. -->
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

sjwiesman commented on doc/user/content/integrations/llm.md:
````diff
@@ -0,0 +1,101 @@
+
+Install and launch with:
+
+```bash
+$ uv pip install mcp-materialize
+$ uv run mcp-materialize
+```
````
This won't work until the deploy PR is merged. Will come back and validate these steps once I can. 

https://github.com/MaterializeInc/materialize/pull/32411

kay-kim replied:
Should they create a venv first? Running the above returns:

```
error: No virtual environment found; run `uv venv` to create an environment, or pass `--system` to install into a non-virtual environment
error: Failed to spawn: `mcp-materialize`
  Caused by: No such file or directory (os error 2)
```

sjwiesman commented:
Link to launch post for more context. https://docs.google.com/document/d/14_9ulwflIwc83b_4pj4dOevrdsNNdJomTZUK61vrzM8/edit?tab=t.0#heading=h.nxlm36x396nq

kay-kim commented on doc/user/content/integrations/llm.md:
````diff
@@ -0,0 +1,101 @@
+* Have `SELECT` privileges on the view.
+* Have `USAGE` privileges on the cluster where the index is installed.
+
+```mzsql
````
Should we include a USAGE on the database and schema here as well by doing something like:

For example, the `payment_status_summary` view is in `materialize.public` (database `materialize` and schema `public`) and the created index is on the `mcp_cluster`:

```mzsql

GRANT USAGE on DATABASE materialize TO mcp_server_role;
GRANT USAGE on SCHEMA materialize.public TO mcp_server_role;
...
```

kay-kim commented on doc/user/content/integrations/llm.md:
```diff
@@ -0,0 +1,101 @@
+* [CREATE VIEW](/sql/create-view)
+* [CREATE INDEX](/sql/create-index)
+* [COMMENT ON](/sql/comment-on)
+* [GRANT PRIVILEGE](/sql/grant-privilege)
```
Maybe add [CREATE ROLE] as well?

sjwiesman commented on doc/user/content/integrations/llm.md:
```diff
@@ -0,0 +1,101 @@
+### 2. Document
+
+The view must include a top-level comment.
+This comment is used as the tool’s description and helps the model reason about what the tool does and when to use it.
```
Not sure I follow this one, but I updated the language to be clearer.

kay-kim replied:
Oh .. .heh heh ... I just meant change the  `’` to `'`.  Sorry for the cryptic shorthand above :D

kay-kim reviewed (approved):
lgtm.  Should we list it on the What's new block to promote the MCP Server for Materialize  https://preview.materialize.com/materialize/32426/ ?

sjwiesman commented:
Yes! Will update that with the launch post when it’s live

kay-kim commented on doc/user/content/integrations/llm.md:
````diff
@@ -0,0 +1,114 @@
+If you don't have uv installed, you can install it first:
+
+```bash
+$ curl -LsSf https://astral.sh/uv/install.sh | sh
````
Oh ... just realized we should delete the prompt `$` -- the copy button copies it

kay-kim commented on doc/user/content/integrations/llm.md:
````diff
@@ -0,0 +1,114 @@
+
+```bash
+$ uv pip install mcp-materialize
+$ uv run mcp-materialize
````
ditto w.r.t. the `$`

sjwiesman commented:
@kay-kim thanks for the reviews! I've updated the page with validated installation steps and added a callout in Whats New. It includes a link to what will be the launch post.

kay-kim commented:
@sjwiesman  -- sounds good.  Feel free to merge this whenever you launch the post.  As a one last tweak (and feel free to ignore) ... since our docs account for ~40% of our site traffic, we could also add a link to your blog post on the llm.md page as well.
